src/components/Message/hooks/useUserRole.ts

```diff
@@ -24,7 +24,9 @@ export const useUserRole = <
   const isModerator =
     client.user?.role === 'channel_moderator' ||
     channel.state.membership.role === 'channel_moderator' ||
-    channel.state.membership.role === 'moderator';
+    channel.state.membership.role === 'moderator' ||
+    channel.state.membership.is_moderator === true ||
+    channel.state.membership.channel_role === 'channel_moderator';
```

Comment:
While this is true, there are multiple things to keep in  mind:

1. `role` is deprecated and we should opt out from using it
2. When v2 of permissions is used, custom roles may be assigned to channel role, making these role checks fragile
3. When v2 of permissions is used, any extra moderator-level role can be added, so it's not binary

I'm OK with this change as long as we put a deprecation notice for this const. We should opt out from any role comparisons in the SDK and rely on provided capabilities instead.

@oliverlaz @petyosi , how is `isModerator` used in core SDK?

@rvanlaarhoven how do you use `isModerator` check? Do you use v2 of permissions?

Thanks

Reply:
Yes, I'm using the v2 permissions.

I'd be happy to update the pull request. I left the `role` in there for backwards compatibility. Just let me know how to address the issue properly to match all the requirements to indicate a user as moderator. 

But since you say these role checks are fragile because of the ability to assign custom roles, it might make sense to deprecate `isModerator` completely to avoid confusion. I had to dig into the source to see what's going on and probably other people reading about the availability of `isModerator` will too.

Reply:
hi @rvanlaarhoven, In order to speed up the process, I added some deprecation comments to this PR as discussed for both `isModerator` and `isAdmin` properties (both of them rely on the already deprecated `membership.role` property). With this, we are ready to merge this PR.

However, we would like to hear more about your use case to use `isModerator` property. Is there something that can't be achieved with `channelCapabilities`?

Reply:
hi @rvanlaarhoven, thanks for sharing the info. `channelCapabilities` is the way to go for sure. I'm happy you can achieve the desired behavior with it.